### PR TITLE
Makefile

### PR DESCRIPTION
**Pull request by janmazak**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,205 +15,182 @@
...
+
+########################################
+#       Additional configuration       #
+########################################
+
+# USB U2F
+DEFINES += HAVE_U2F HAVE_IO_U2F U2F_PROXY_MAGIC=\"ADA\"
+SDK_SOURCE_PATH  += lib_u2f
+
+DEFINES += HAVE_UX_LEGACY
```
Comment (xchapron-ledger) [inline Makefile:153]: This wasn't included in the previous version, is it really needed?

File: Makefile [hunk 2]
```diff
@@ -15,205 +15,182 @@
...
 
-NANOS_ID = 1
+# mnemonic and PIN for testing on a physical device
 WORDS = "abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about"
 PIN = 5555
 
-APP_LOAD_PARAMS =--appFlags 0x200 --curve ed25519 --path "44'/1815'" --path "1852'/1815'" --path "1853'/1815'" --path "1854'/1815'" --path "1855'/1815'" --path "1694'/1815'"
-APP_LOAD_PARAMS += $(COMMON_LOAD_PARAMS)
+# based in part on https://interrupt.memfault.com/blog/best-and-worst-gcc-clang-compiler-flags
+WERROR   := -Werror=return-type -Werror=parentheses -Werror=format-security
```
Comment (xchapron-ledger) [inline Makefile:163]: `-Werror=format-security` this warning is already enabled in the SDK.
Regarding the whole line, I would recommend adding theses as warning, which are converted as errors only for CI builds.

File: Makefile [hunk 3]
```diff
@@ -15,205 +15,182 @@
...
-	python -m ledgerblue.loadApp $(APP_LOAD_PARAMS)
+########################################
+#          Additional targets          #
+########################################
 
-delete:
-	python -m ledgerblue.deleteApp $(COMMON_DELETE_PARAMS)
+# code style
+format:
+	astyle --options=.astylerc "src/*.h" "src/*.c" --exclude=src/glyphs.h --exclude=src/glyphs.c --ignore-exclude-errors
```
Comment (xchapron-ledger) [inline Makefile:172]: glyphs.c and glyphs.h are not generated in src anymore.
You could make a CI check of the coding style to make sure code stay formatted.

File: Makefile [hunk 4]
```diff
@@ -15,205 +15,182 @@
...
 
+stax: clean
+	BOLOS_SDK=$(STAX_SDK) make
 
-##############
-#    Size    #
-##############
+# part of CI
+analyze: clean
+	scan-build --use-cc=clang -analyze-headers -enable-checker security -enable-checker unix -enable-checker valist -o scan-build --status-bugs make default
```
Comment (xchapron-ledger) [inline Makefile:193]: Could be replaced by a simpler call to `make scan-build`?

File: Makefile (hunk 1 above)
Comment (xchapron-ledger) [reply Makefile:153]: I think what you need is something like:
```
ifeq ($(TARGET_NAME), TARGET_NANOS)
    DISABLE_STANDARD_BAGL_UX_FLOW=1
endif
```
To keep the same kind of stuff that you had before.
Though dropping the usage of old nanos ux code, to use the common one would be an interssting improvement.
I see that it's automatically done if you have HAVE_FLOW in uiHelpers_nanos.c and uiHelpers_nanox.c but not in menu_nanos.c, menu_nanox.c, menu.h, io.c, uiHelpers.h, io.h, uiScreens_bagl.c, uiHelpers.c

File: src/cardano_io.c
Comment (xchapron-ledger) [inline src/cardano_io.c:1]: Could probably be really simplified if using UX_FLOW on TARGET_NANOS

File: Makefile (hunk 1 above)
Comment (janmazak) [reply Makefile]: I've tried it, however, I'd say that `lib_standard_app/main.c` is not compatible with it:
```
/opt/nanos-secure-sdk/lib_standard_app/main.c:33:19: error: redefinition of 'ux' with a different type: 'bolos_ux_params_t' (aka 'struct bolos_ux_params_s') vs 'ux_state_t' (aka 'struct ux_state_s')
bolos_ux_params_t G_ux_params;
                  ^
/opt/nanos-secure-sdk/lib_ux/include/ux.h:322:30: note: expanded from macro 'G_ux_params'
#define G_ux_params          ux.params
                             ^
/opt/nanos-secure-sdk/lib_standard_app/main.c:32:19: note: previous definition is here
ux_state_t        G_ux;
                  ^
/opt/nanos-secure-sdk/lib_ux/include/ux.h:321:30: note: expanded from macro 'G_ux'
#define G_ux                 ux
                             ^
/opt/nanos-secure-sdk/lib_standard_app/main.c:33:19: error: expected ';' after top level declarator
bolos_ux_params_t G_ux_params;
                  ^
/opt/nanos-secure-sdk/lib_ux/include/ux.h:322:32: note: expanded from macro 'G_ux_params'
#define G_ux_params          ux.params
```
@xchapron-ledger Any advice, please?

Comment (janmazak) [reply Makefile]: (At some point, we might rewrite the UI to use the new version, but at present we don't have time and budget for that.)

Comment (xchapron-ledger) [reply Makefile]: As you want, I'm even wondering if you have anything to do.
I looks like you could simply:
- Enable HAVE_FLOW for TARGET_NANOS
- Update some #ifdef in menu_nanos.c, menu_nanox.c, menu.h, io.c, uiHelpers.h, io.h, uiScreens_bagl.c, uiHelpers.c
And that should be all?
But could be for another time.

**Comment by xchapron-ledger** [comment]
The Makefile refactoring part (using `makefile.standard_app`) was interesting to have. Do you intend to drop it or do you want to push it differently?

**Comment by janmazak** [comment]
This got closed automatically, but we've not given up on Makefile refactoring.
https://github.com/vacuumlabs/ledger-app-cardano-shelley/pull/210
It will be part of the larger UX update planned by Ledger in 1H24 (so that "swap" can be implemented).

**Pull request closed without merging**